pypreprocess/nipype_preproc_spm_utils.py

```diff
@@ -193,8 +192,9 @@ def _do_subject_slice_timing(subject_data, TR, TA=None, spm_dir=None,
     for sess_func in subject_data.func:
         stc_result = stc(in_files=sess_func, time_repetition=TR,
                          time_acquisition=TA, num_slices=nslices,
-                         ref_slice=ref_slice + 1,
-                         slice_order=list(slice_order + 1),  # SPM
+                         ref_slice=ref_slice,
+                         #slice_order=list(slice_order + 1),  # SPM8 ?
+                         slice_order=[int(x) for x in slice_order.split()],  # SPM12
```

Comment:
* This breaks things. Here you're assuming that slicer_order is a list space-separated string of integers like "1 2 3 4 5". What if it's a list like [1, 2, 3, 4, 5] ? What if it's a string like "ascending" (with the usual meaning) ?
* The index-generation logic should be factored out and relocated in a testable function like slice_timing.get_indices. The later has been corrected for the bug you spotted (double computing of indices, etc.) and so this refactoring should be smooth.